aarch64-linux build fails ( tested with virtualisation)
`nix build ".#tinycc" --system aarch64-linux`
```
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/rzk09xlx2affxxqjws8km36ymmk6mfsm-source
source root is source
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
configure flags: --disable-static --prefix=/nix/store/q7qvb3wm7nharlsmp2qsjc3s6kd0526y-tcc-unstable-2021-07-27 --cc=cc --elfinterp=/nix/store/34k9b4lsmr7mcmykvbmwjazydwnfkckk-glibc-2.33-50/lib/ld-linux-aarch64.so.1 --crtprefix=/nix/store/34k9b4lsmr7mcmykvbmwjazydwnfkckk-glibc-2.33-50/lib --sysincludepaths=/nix/store/vlb82s8dpdrllzgbk0q75n433lam3q0z-glibc-2.33-50-dev/include:\{B\}/include --libpaths=/nix/store/34k9b4lsmr7mcmykvbmwjazydwnfkckk-glibc-2.33-50/lib
Binary directory    /nix/store/q7qvb3wm7nharlsmp2qsjc3s6kd0526y-tcc-unstable-2021-07-27/bin
TinyCC directory    /nix/store/q7qvb3wm7nharlsmp2qsjc3s6kd0526y-tcc-unstable-2021-07-27/lib/tcc
Library directory   /nix/store/q7qvb3wm7nharlsmp2qsjc3s6kd0526y-tcc-unstable-2021-07-27/lib
Include directory   /nix/store/q7qvb3wm7nharlsmp2qsjc3s6kd0526y-tcc-unstable-2021-07-27/include
Manual directory    /nix/store/q7qvb3wm7nharlsmp2qsjc3s6kd0526y-tcc-unstable-2021-07-27/share/man
Info directory      /nix/store/q7qvb3wm7nharlsmp2qsjc3s6kd0526y-tcc-unstable-2021-07-27/share/info
Doc directory       /nix/store/q7qvb3wm7nharlsmp2qsjc3s6kd0526y-tcc-unstable-2021-07-27/share/doc
Source path         /build/source
C compiler          cc (9.3)
Target OS           Linux
CPU                 arm64
Config              static=no
Creating config.mak and config.h
@nix { "action": "setPhase", "phase": "buildPhase" }
building
build flags: -j24 -l24 SHELL=/nix/store/b9kazb4rcz4mm7am51c8zbzqxl4dycr1-bash-5.1-p8/bin/bash
cc -o tcc.o -c tcc.c -DCONFIG_TCC_SYSINCLUDEPATHS="\"/nix/store/vlb82s8dpdrllzgbk0q75n433lam3q0z-glibc-2.33-50-dev/include:{B}/include\"" -DCONFIG_TCC_LIBPATHS="\"/nix/store/34k9b4lsmr7mcmykvbmwjazydwnfkckk-glibc-2.33-50/lib\"" -DCONFIG_TCC_CRTPREFIX="\"/nix/store/34k9b4lsmr7mcmykvbmwjazydwnfkckk-glibc-2.33-50/lib\"" -DCONFIG_TCC_ELFINTERP="\"/nix/store/34k9b4lsmr7mcmykvbmwjazydwnfkckk-glibc-2.33-50/lib/ld-linux-aarch64.so.1\"" -DTCC_TARGET_ARM64        -DONE_SOURCE=0  -Wall -O2 -Wdeclaration-after-statement -fno-strict-aliasing -Wno-pointer-sign -Wno-sign-compare -Wno-unused-result -Wno-format-truncation -I. 
cc -o libtcc.o -c libtcc.c -DCONFIG_TCC_SYSINCLUDEPATHS="\"/nix/store/vlb82s8dpdrllzgbk0q75n433lam3q0z-glibc-2.33-50-dev/include:{B}/include\"" -DCONFIG_TCC_LIBPATHS="\"/nix/store/34k9b4lsmr7mcmykvbmwjazydwnfkckk-glibc-2.33-50/lib\"" -DCONFIG_TCC_CRTPREFIX="\"/nix/store/34k9b4lsmr7mcmykvbmwjazydwnfkckk-glibc-2.33-50/lib\"" -DCONFIG_TCC_ELFINTERP="\"/nix/store/34k9b4lsmr7mcmykvbmwjazydwnfkckk-glibc-2.33-50/lib/ld-linux-aarch64.so.1\"" -DTCC_TARGET_ARM64        -DONE_SOURCE=0 -Wall -O2 -Wdeclaration-after-statement -fno-strict-aliasing -Wno-pointer-sign -Wno-sign-compare -Wno-unused-result -Wno-format-truncation -I.  -fPIC
cc -DC2STR conftest.c -o c2str.exe && ./c2str.exe include/tccdefs.h tccdefs_.h
cc -o tccgen.o -c tccgen.c -DCONFIG_TCC_SYSINCLUDEPATHS="\"/nix/store/vlb82s8dpdrllzgbk0q75n433lam3q0z-glibc-2.33-50-dev/include:{B}/include\"" -DCONFIG_TCC_LIBPATHS="\"/nix/store/34k9b4lsmr7mcmykvbmwjazydwnfkckk-glibc-2.33-50/lib\"" -DCONFIG_TCC_CRTPREFIX="\"/nix/store/34k9b4lsmr7mcmykvbmwjazydwnfkckk-glibc-2.33-50/lib\"" -DCONFIG_TCC_ELFINTERP="\"/nix/store/34k9b4lsmr7mcmykvbmwjazydwnfkckk-glibc-2.33-50/lib/ld-linux-aarch64.so.1\"" -DTCC_TARGET_ARM64        -DONE_SOURCE=0 -Wall -O2 -Wdeclaration-after-statement -fno-strict-aliasing -Wno-pointer-sign -Wno-sign-compare -Wno-unused-result -Wno-format-truncation -I.  -fPIC
cc -o tccelf.o -c tccelf.c -DCONFIG_TCC_SYSINCLUDEPATHS="\"/nix/store/vlb82s8dpdrllzgbk0q75n433lam3q0z-glibc-2.33-50-dev/include:{B}/include\"" -DCONFIG_TCC_LIBPATHS="\"/nix/store/34k9b4lsmr7mcmykvbmwjazydwnfkckk-glibc-2.33-50/lib\"" -DCONFIG_TCC_CRTPREFIX="\"/nix/store/34k9b4lsmr7mcmykvbmwjazydwnfkckk-glibc-2.33-50/lib\"" -DCONFIG_TCC_ELFINTERP="\"/nix/store/34k9b4lsmr7mcmykvbmwjazydwnfkckk-glibc-2.33-50/lib/ld-linux-aarch64.so.1\"" -DTCC_TARGET_ARM64        -DONE_SOURCE=0 -Wall -O2 -Wdeclaration-after-statement -fno-strict-aliasing -Wno-pointer-sign -Wno-sign-compare -Wno-unused-result -Wno-format-truncation -I.  -fPIC
cc -o tccasm.o -c tccasm.c -DCONFIG_TCC_SYSINCLUDEPATHS="\"/nix/store/vlb82s8dpdrllzgbk0q75n433lam3q0z-glibc-2.33-50-dev/include:{B}/include\"" -DCONFIG_TCC_LIBPATHS="\"/nix/store/34k9b4lsmr7mcmykvbmwjazydwnfkckk-glibc-2.33-50/lib\"" -DCONFIG_TCC_CRTPREFIX="\"/nix/store/34k9b4lsmr7mcmykvbmwjazydwnfkckk-glibc-2.33-50/lib\"" -DCONFIG_TCC_ELFINTERP="\"/nix/store/34k9b4lsmr7mcmykvbmwjazydwnfkckk-glibc-2.33-50/lib/ld-linux-aarch64.so.1\"" -DTCC_TARGET_ARM64        -DONE_SOURCE=0 -Wall -O2 -Wdeclaration-after-statement -fno-strict-aliasing -Wno-pointer-sign -Wno-sign-compare -Wno-unused-result -Wno-format-truncation -I.  -fPIC
cc -o tccrun.o -c tccrun.c -DCONFIG_TCC_SYSINCLUDEPATHS="\"/nix/store/vlb82s8dpdrllzgbk0q75n433lam3q0z-glibc-2.33-50-dev/include:{B}/include\"" -DCONFIG_TCC_LIBPATHS="\"/nix/store/34k9b4lsmr7mcmykvbmwjazydwnfkckk-glibc-2.33-50/lib\"" -DCONFIG_TCC_CRTPREFIX="\"/nix/store/34k9b4lsmr7mcmykvbmwjazydwnfkckk-glibc-2.33-50/lib\"" -DCONFIG_TCC_ELFINTERP="\"/nix/store/34k9b4lsmr7mcmykvbmwjazydwnfkckk-glibc-2.33-50/lib/ld-linux-aarch64.so.1\"" -DTCC_TARGET_ARM64        -DONE_SOURCE=0 -Wall -O2 -Wdeclaration-after-statement -fno-strict-aliasing -Wno-pointer-sign -Wno-sign-compare -Wno-unused-result -Wno-format-truncation -I.  -fPIC
cc -o arm64-gen.o -c arm64-gen.c -DCONFIG_TCC_SYSINCLUDEPATHS="\"/nix/store/vlb82s8dpdrllzgbk0q75n433lam3q0z-glibc-2.33-50-dev/include:{B}/include\"" -DCONFIG_TCC_LIBPATHS="\"/nix/store/34k9b4lsmr7mcmykvbmwjazydwnfkckk-glibc-2.33-50/lib\"" -DCONFIG_TCC_CRTPREFIX="\"/nix/store/34k9b4lsmr7mcmykvbmwjazydwnfkckk-glibc-2.33-50/lib\"" -DCONFIG_TCC_ELFINTERP="\"/nix/store/34k9b4lsmr7mcmykvbmwjazydwnfkckk-glibc-2.33-50/lib/ld-linux-aarch64.so.1\"" -DTCC_TARGET_ARM64        -DONE_SOURCE=0 -Wall -O2 -Wdeclaration-after-statement -fno-strict-aliasing -Wno-pointer-sign -Wno-sign-compare -Wno-unused-result -Wno-format-truncation -I.  -fPIC
cc -o arm64-link.o -c arm64-link.c -DCONFIG_TCC_SYSINCLUDEPATHS="\"/nix/store/vlb82s8dpdrllzgbk0q75n433lam3q0z-glibc-2.33-50-dev/include:{B}/include\"" -DCONFIG_TCC_LIBPATHS="\"/nix/store/34k9b4lsmr7mcmykvbmwjazydwnfkckk-glibc-2.33-50/lib\"" -DCONFIG_TCC_CRTPREFIX="\"/nix/store/34k9b4lsmr7mcmykvbmwjazydwnfkckk-glibc-2.33-50/lib\"" -DCONFIG_TCC_ELFINTERP="\"/nix/store/34k9b4lsmr7mcmykvbmwjazydwnfkckk-glibc-2.33-50/lib/ld-linux-aarch64.so.1\"" -DTCC_TARGET_ARM64        -DONE_SOURCE=0 -Wall -O2 -Wdeclaration-after-statement -fno-strict-aliasing -Wno-pointer-sign -Wno-sign-compare -Wno-unused-result -Wno-format-truncation -I.  -fPIC
cc -o arm64-asm.o -c arm64-asm.c -DCONFIG_TCC_SYSINCLUDEPATHS="\"/nix/store/vlb82s8dpdrllzgbk0q75n433lam3q0z-glibc-2.33-50-dev/include:{B}/include\"" -DCONFIG_TCC_LIBPATHS="\"/nix/store/34k9b4lsmr7mcmykvbmwjazydwnfkckk-glibc-2.33-50/lib\"" -DCONFIG_TCC_CRTPREFIX="\"/nix/store/34k9b4lsmr7mcmykvbmwjazydwnfkckk-glibc-2.33-50/lib\"" -DCONFIG_TCC_ELFINTERP="\"/nix/store/34k9b4lsmr7mcmykvbmwjazydwnfkckk-glibc-2.33-50/lib/ld-linux-aarch64.so.1\"" -DTCC_TARGET_ARM64        -DONE_SOURCE=0 -Wall -O2 -Wdeclaration-after-statement -fno-strict-aliasing -Wno-pointer-sign -Wno-sign-compare -Wno-unused-result -Wno-format-truncation -I.  -fPIC
perl ./texi2pod.pl tcc-doc.texi tcc-doc.pod
makeinfo --no-split --html --number-sections -o tcc-doc.html tcc-doc.texi
makeinfo tcc-doc.texi || true
pod2man --section=1 --center="Tiny C Compiler" --release="unstable-2021-07-27" tcc-doc.pod >tcc.1 && rm -f tcc-doc.pod
cc -o tccpp.o -c tccpp.c -DCONFIG_TCC_SYSINCLUDEPATHS="\"/nix/store/vlb82s8dpdrllzgbk0q75n433lam3q0z-glibc-2.33-50-dev/include:{B}/include\"" -DCONFIG_TCC_LIBPATHS="\"/nix/store/34k9b4lsmr7mcmykvbmwjazydwnfkckk-glibc-2.33-50/lib\"" -DCONFIG_TCC_CRTPREFIX="\"/nix/store/34k9b4lsmr7mcmykvbmwjazydwnfkckk-glibc-2.33-50/lib\"" -DCONFIG_TCC_ELFINTERP="\"/nix/store/34k9b4lsmr7mcmykvbmwjazydwnfkckk-glibc-2.33-50/lib/ld-linux-aarch64.so.1\"" -DTCC_TARGET_ARM64        -DONE_SOURCE=0 -Wall -O2 -Wdeclaration-after-statement -fno-strict-aliasing -Wno-pointer-sign -Wno-sign-compare -Wno-unused-result -Wno-format-truncation -I.  -fPIC
cc -shared -Wl,-soname,libtcc.so -o libtcc.so libtcc.o tccpp.o tccgen.o tccelf.o tccasm.o tccrun.o arm64-gen.o arm64-link.o arm64-asm.o -s -fPIC
cc -o tcc tcc.o libtcc.so -lm -lpthread -ldl -s -Wl,-rpath,"/nix/store/q7qvb3wm7nharlsmp2qsjc3s6kd0526y-tcc-unstable-2021-07-27/lib"
make[1]: Entering directory '/build/source/lib'
../tcc -c lib-arm64.c -o lib-arm64.o -B.. -I..
../tcc -c fetch_and_add_arm64.S -o fetch_and_add_arm64.o -B.. -I..
../tcc -c tcov.c -o tcov.o -B.. -I..
../tcc -c dsohandle.c -o dsohandle.o -B.. -I..
../tcc -c bt-exe.c -o ../bt-exe.o -B.. -I..
../tcc -c bt-log.c -o ../bt-log.o -B.. -I..
../tcc -c bcheck.c -o ../bcheck.o -B.. -I.. -g 
../tcc -ar rcs ../libtcc1.a lib-arm64.o fetch_and_add_arm64.o tcov.o dsohandle.o
make[1]: Leaving directory '/build/source/lib'
@nix { "action": "setPhase", "phase": "checkPhase" }
running tests
check flags: -j24 -l24 SHELL=/nix/store/b9kazb4rcz4mm7am51c8zbzqxl4dycr1-bash-5.1-p8/bin/bash VERBOSE=y test
make[1]: Entering directory '/build/source/tests'
------------ hello-exe ------------
------------ memtest ------------
------------ dlltest ------------
------------ hello-run ------------
------------ cross-test ------------
------------ pp-dir ------------
------------ btest ------------
------------ tests2-dir ------------
 . i386
Hello World
------------ vla_test-run ------------
Hello World
PPTest 03 ...
PPTest 01 ...
Test 1 succeeded as expected
PPTest 06 ...
PPTest 02 ...
PPTest 12 ...
PPTest 04 ...
PPTest 08 ...
PPTest 09 ...
PPTest 11 ...
PPTest 05 ...
PPTest 07 ...
PPTest 10 ...
Test: 00_assignment...
PPTest 13 ...
test1... success
test2... success
test3... success
PPTest 14 ...
Test 4 succeeded as expected
Test: 01_comment...
Test 8 succeeded as expected
Test: 02_printf...
Test: 03_struct...
Test: 04_for...
Test: 06_case...
Test: 05_array...
PPTest 15 ...
Test: 07_function...
Test: 08_while...
Test: 09_do_while...
PPTest 17 ...
Test: 10_pointer...
PPTest 16 ...
PPTest 18 ...
Test: 11_precedence...
Test 10 succeeded as expected
PPTest 19 ...
PPTest 20 ...
Test 14 succeeded as expected
Test: 12_hashdefine...
Test: 13_integer_literals...
PPTest 21 ...
PPTest pp-counter ...
Test: 15_recursion...
Test: 14_if...
Test: 16_nesting...
Test: 17_enum...
------------ abitest-tcc ------------
Test: 19_pointer_arithmetic...
Test: 18_include...
Test: 20_pointer_comparison...
Test: 21_char_array...
Test: 22_floating_point...
Test 16 succeeded as expected
ret_int_test... success
ret_longlong_test...  . i386-win32
success
ret_float_test... success
ret_double_test... success
ret_longdouble_test... success
ret_2float_test... success
ret_2double_test... success
ret_8plus2double_test... success
ret_6plus2longlong_test... Test 2 failed as expected
success
ret_mixed_test... success
ret_mixed2_test... success
ret_mixed3_test... success
reg_pack_test... success
reg_pack_longlong_test... success
sret_test... Test: 23_type_coercion...
success
one_member_union_test... Test: 24_math_library...
success
two_member_union_test... success
many_struct_test... Test: 25_quicksort...
success
many_struct_test_2... success
many_struct_test_3... Test: 26_character_constants...
success
stdarg_test... Test: 27_sizeof...
Test: 29_array_address...
Test: 28_strings...
success
stdarg_many_test... success
stdarg_struct_test... Test: 30_hanoi...
success
arg_align_test... success
Test: 31_args...
Test: 32_led...
Test: 33_ternary_op...
Test: 35_sizeof...
Test: 36_array_initialisers...
Test 5 failed as expected
Test: 37_sprintf...
Test: 38_multiple_array_index...
Test: 39_typedef...
Test 6 failed as expected
Test: 40_stdio...
Test: 41_hashif...
Test: 43_void_param...
Test: 44_scoped_declarations...
Test: 42_function_pointer...
Test: 46_grep...
Test: 45_empty_for...
Test 7 failed as expected
Test: 47_switch_return...
Test: 48_nested_break...
Test: 51_static...
Test: 50_logical_second_arg...
Test: 49_bracket_evaluation...
Test: 52_unnamed_enum...
Test: 54_goto...
Test 9 failed as expected
Test: 60_errors_and_warnings...
Test: 55_lshift_type...
Test 11 failed as expected
Test: 64_macro_nesting...
Test: 61_integers...
Test: 67_macro_concat...
Test: 70_floating_point_literals...
Test: 72_long_long_constant...
Test: 71_macro_empty_arg...
Test: 73_arm64...
Test: 76_dollars_in_identifiers...
Test: 75_array_in_struct_init...
------------ libtest ------------
Test: 77_push_pop_macro...
Test: 78_vla_label...
Test: 79_vla_continue...
Test: 80_flexarray...
Test 12 failed as expected
Test: 81_types...
Test: 82_attribs_position...
Hello World!
 . i386-OpenBSD
Test 13 failed as expected
Test: 83_utf8_in_identifiers...
------------ libtest_mt ------------
Test: 84_hex-float...
Test: 87_dead_code...
Test: 86_memory-model...
Test: 88_codeopt...
Test: 89_nocode_wanted...
Test: 90_struct-init...
fib(32) = 2178309
add(32, 64) = 96
Test 15 failed as expected
running fib with mixed calls
 Test: 91_ptr_longlong_arith32...
Test: 92_enum_bitfield...
Test: 93_integer_promotion...
Test: 95_bitfields...
Test: 94_generic...
Test: 95_bitfields_ms...
Test: 97_utf8_string_literal...
Test: 100_c99array-decls...
Test: 96_nodata_wanted...
 1 2 3 5 8 13 21 34 55 89 144 233 377 610 987 1597 2584 4181 6765 10946
 (136 ms)
running fib in threads
 Test: 101_cleanup...
Test 17 failed as expected
Test: 102_alignas...
Test: 103_implicit_memmove...
Test: 104_inline...
Hello World
Test: 105_local_extern...
------------ abitest-cc ------------
Test: 106_versym...
Test 18 failed as expected
Bound test OK
Test: 107_stack_safe...
 2 1 5 3 8 13 21 34 55 89 144 233 610 377 987 1597 2584 4181 6765 10946
 (170 ms)
running tcc.c in threads to run fib
 Test: 109_float_struct_calling...
------------ dlltest with PIC ------------
Test: 110_average...
Test: 108_constructor...
Test: 113_btdll...
Test: 111_conversion...
Test: 112_backtrace...
Test: 114_bound_signal...
Test: 115_bound_setjmp...
ret_int_test... Test: 116_bound_setjmp2...
success
ret_longlong_test... success
ret_float_test... success
ret_double_test... Test: 118_switch...
success
ret_longdouble_test... Test: 117_builtins...
success
ret_2float_test... success
ret_2double_test... success
ret_8plus2double_test... success
ret_6plus2longlong_test... success
ret_mixed_test... success
ret_mixed2_test... success
ret_mixed3_test... success
reg_pack_test... success
reg_pack_longlong_test... success
sret_test... success
one_member_union_test... success
two_member_union_test... success
many_struct_test... success
many_struct_test_2... success
many_struct_test_3... success
stdarg_test... success
stdarg_many_test... Test: 119_random_stuff...
success
stdarg_struct_test... success
arg_align_test... success
Test: 120_alias...
Test: 122_vla_reuse...
Test: 121_struct_return...
--- 108_constructor.expect	1970-01-01 00:00:01.000000000 +0000
+++ 108_constructor.output	2021-10-10 18:09:22.738925881 +0000
Test: 123_vla_bug...
@@ -1,3 +1 @@
-constructor
-main
-destructor
+ok
make[3]: *** [Makefile:132: 108_constructor.test] Error 1
 . x86_64
In file included from [01m[K../libtcc.c:25[m[K,
                 from [01m[K../tcc.c:23[m[K:
[01m[K../tccelf.c:[m[K In function '[01m[Kld_add_file_list[m[K':
[01m[K../tccelf.c:3739:57:[m[K [01;35m[Kwarning: [m[K'[01m[K%s[m[K' directive output may be truncated writing up to 1023 bytes into a region of size 1021 [[01;35m[K-Wformat-truncation=[m[K]
 3739 |                 snprintf(filename, sizeof filename, "lib[01;35m[K%s[m[K.so", [32m[Klibname[m[K);
      |                                                         [01;35m[K^~[m[K      [32m[K~~~~~~~[m[K
[01m[K../tccelf.c:3739:17:[m[K [01;36m[Knote: [m[K'[01m[Ksnprintf[m[K' output between 7 and 1030 bytes into a destination of size 1024
 3739 |                 [01;36m[Ksnprintf(filename, sizeof filename, "lib%s.so", libname)[m[K;
      |                 [01;36m[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~[m[K
[01m[K../tccelf.c:3737:57:[m[K [01;35m[Kwarning: [m[K'[01m[K%s[m[K' directive output may be truncated writing up to 1023 bytes into a region of size 1021 [[01;35m[K-Wformat-truncation=[m[K]
 3737 |                 snprintf(filename, sizeof filename, "lib[01;35m[K%s[m[K.a", [32m[Klibname[m[K);
      |                                                         [01;35m[K^~[m[K     [32m[K~~~~~~~[m[K
[01m[K../tccelf.c:3737:17:[m[K [01;36m[Knote: [m[K'[01m[Ksnprintf[m[K' output between 6 and 1029 bytes into a destination of size 1024
 3737 |                 [01;36m[Ksnprintf(filename, sizeof filename, "lib%s.a", libname)[m[K;
      |                 [01;36m[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~[m[K
 2 1 . x86_64-win32
 3------------ test3 ------------
------------ test1b ------------
In file included from [01m[K../libtcc.c:25[m[K:
[01m[K../tccelf.c:[m[K In function '[01m[Kld_add_file_list[m[K':
[01m[K../tccelf.c:3739:57:[m[K [01;35m[Kwarning: [m[K'[01m[K%s[m[K' directive output may be truncated writing up to 1023 bytes into a region of size 1021 [[01;35m[K-Wformat-truncation=[m[K]
 3739 |                 snprintf(filename, sizeof filename, "lib[01;35m[K%s[m[K.so", [32m[Klibname[m[K);
      |                                                         [01;35m[K^~[m[K      [32m[K~~~~~~~[m[K
[01m[K../tccelf.c:3739:17:[m[K [01;36m[Knote: [m[K'[01m[Ksnprintf[m[K' output between 7 and 1030 bytes into a destination of size 1024
 3739 |                 [01;36m[Ksnprintf(filename, sizeof filename, "lib%s.so", libname)[m[K;
      |                 [01;36m[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~[m[K
[01m[K../tccelf.c:3737:57:[m[K [01;35m[Kwarning: [m[K'[01m[K%s[m[K' directive output may be truncated writing up to 1023 bytes into a region of size 1021 [[01;35m[K-Wformat-truncation=[m[K]
 3737 |                 snprintf(filename, sizeof filename, "lib[01;35m[K%s[m[K.a", [32m[Klibname[m[K);
      |                                                         [01;35m[K^~[m[K     [32m[K~~~~~~~[m[K
[01m[K../tccelf.c:3737:17:[m[K [01;36m[Knote: [m[K'[01m[Ksnprintf[m[K' output between 6 and 1029 bytes into a destination of size 1024
 3737 |                 [01;36m[Ksnprintf(filename, sizeof filename, "lib%s.a", libname)[m[K;
      |                 [01;36m[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~[m[K
make[3]: Target 'all' not remade because of errors.
make[2]: *** [Makefile:104: tests2-dir] Error 2
make[2]: *** Waiting for unfinished jobs....
 . x86_64-osx
 5 8 . x86_64-FreeBSD
 13 . x86_64-NetBSD
 21 . x86_64-OpenBSD
 34Auto Test3 OK
 55 . arm-fpa
 89 144Auto Bound-Test OK
 . arm-eabihf
 233 377 . arm-NetBSD
 610 987 . arm-wince
 1597 2584 . arm64
 4181 6765 . arm64-osx
 10946
 (5879 ms)
compiling tcc.c 10 times
  1 2 . arm64-FreeBSD
 3Hello World
OK
 4 . arm64-NetBSD
 5 6 . arm64-OpenBSD
 7 8 . riscv64
 9 10 . c67

 (2178 ms)
make[1]: *** [Makefile:83: all] Error 2
make[1]: Leaving directory '/build/source/tests'
make: *** [Makefile:427: test] Error 2
```

--- a/pkgs/development/compilers/tinycc/default.nix
+++ b/pkgs/development/compilers/tinycc/default.nix
@@ -1,36 +1,35 @@
-{ stdenv, lib, fetchFromRepoOrCz, perl, texinfo }:
-with lib;
+{ stdenv, lib, fetchFromRepoOrCz, perl, texinfo, which }:
 
 stdenv.mkDerivation rec {
   pname = "tcc";
-  version = "0.9.27";
-  upstreamVersion = "release_${concatStringsSep "_" (builtins.splitVersion version)}";
+  version = "unstable-2021-10-09";
 
   src = fetchFromRepoOrCz {
     repo = "tinycc";
-    rev = upstreamVersion;
-    sha256 = "12mm1lqywz0akr2yb2axjfbw8lwv57nh395vzsk534riz03ml977";
+    rev = "ca11849ebb88ef4ff87beda46bf5687e22949bd6";
+    sha256 = "sha256-xnUDyTYZxbxUCblACyX73boBhU073VRqSy1SWlWsvIw=";
   };
 
-  nativeBuildInputs = [ perl texinfo ];
+  nativeBuildInputs = [ perl texinfo which ];
 
   hardeningDisable = [ "fortify" ];
 
-  enableParallelBuilding = true;
-
   postPatch = ''
-    substituteInPlace "texi2pod.pl" \
-      --replace "/usr/bin/perl" "${perl}/bin/perl"
+    patchShebangs texi2pod.pl
   '';
+
+  configureFlags = [
+    "--cc=cc"
+    "--crtprefix=${lib.getLib stdenv.cc.libc}/lib"
+    "--sysincludepaths=${lib.getDev stdenv.cc.libc}/include:{B}/include"
+    "--libpaths=${lib.getLib stdenv.cc.libc}/lib"
+    # build cross compilers
+    "--enable-cross"
+  ];
 
   preConfigure = ''
     echo ${version} > VERSION
-
-    configureFlagsArray+=("--cc=cc")
     configureFlagsArray+=("--elfinterp=$(< $NIX_CC/nix-support/dynamic-linker)")
-    configureFlagsArray+=("--crtprefix=${getLib stdenv.cc.libc}/lib")
-    configureFlagsArray+=("--sysincludepaths=${getDev stdenv.cc.libc}/include:{B}/include")
-    configureFlagsArray+=("--libpaths=${getLib stdenv.cc.libc}/lib")
   '';
 
   postFixup = ''
@@ -47,9 +46,8 @@ stdenv.mkDerivation rec {
   doCheck = true;
   checkTarget = "test";
 
-  meta = {
+  meta = with lib; {
     description = "Small, fast, and embeddable C compiler and interpreter";
-
     longDescription = ''
       TinyCC (aka TCC) is a small but hyper fast C compiler.  Unlike
       other C compilers, it is meant to be self-sufficient: you do not
@@ -73,11 +71,9 @@ stdenv.mkDerivation rec {
       With libtcc, you can use TCC as a backend for dynamic code
       generation.
     '';
-
-    homepage = "http://www.tinycc.org/";
-    license = licenses.mit;
-
-    platforms = [ "x86_64-linux" ];
+    homepage = "https://repo.or.cz/tinycc.git";
+    license = licenses.lgpl21Only;
+    platforms = platforms.linux;
     maintainers = [ maintainers.joachifm ];
   };
 }

